File: nim/str_join_repeat.nim
import times
import strutils

const chunk = "regexp123abcbcd"
const repeatCount = 10000
const size = 1000

proc join(): void =
    discard chunk.repeat(size)

let startTime = epochTime()
for i in 1..repeatCount:
    join()
let endTime = epochTime()

echo int((endTime - startTime) * 1000)

File: lua/str_join_repeat.lua
local chunk = "regexp123abcbcd"
local repeatCount = 10000
local size = 1000

function join()
  chunk:rep(size)
end

local start = os.clock()
for i = 1, repeatCount do
  join()
end
local _end = os.clock()

print((_end - start) * 1000)
